binning-utilities/src/main/java/com/oculusinfo/factory/ConfigurableFactory.java

```diff
@@ -192,7 +192,7 @@ public String getName() {
 	 */
 	public <PT> void addProperty (ConfigurationProperty<PT> property, List<String> path) {
 		_properties.add(property);
-        _pathsByProperty.put( property.getName(), path );
+        _pathsByPropertyUUID.put( property.getUUID(), path );
```

Comment:
Would it be a giant pain instead to change this to: 

    _pathsByProperty.put(property, path)

and put the UUID checks into the property.hashCode and property.equals function?

I know it's a few more places, since there are multiple property types, but I think it makes this code a lot cleaner and less implementation-dependent - i.e., nothing outside the properties needs to know they have a UUID